src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_INITIAL_CAPACITY = 16;
+    public static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    public static final int DEFAULT_ARRAY_EXPANSION = 2;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 > threshold) {
+            resize();
+        }
+        putNode(key, value);
     }
 
     @Override
     public V getValue(K key) {
+        int hash = getIndex(key);
+        Node<K, V> currentNode = table[hash];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        threshold = (int) (table.length * DEFAULT_ARRAY_EXPANSION * DEFAULT_LOAD_FACTOR);
+        Node<K, V>[] oldTable = table;
+        table = new Node[table.length * DEFAULT_ARRAY_EXPANSION];
+        copy(oldTable);
+    }
+
+    private void copy(Node<K, V>[] oldTable) {
+        size = 0;
+        for (Node<K, V> currentNode : oldTable) {
+            while (currentNode != null) {
+                putNode(currentNode.key, currentNode.value);
+                currentNode = currentNode.next;
+            }
+        }
+    }
+
+    private void putNode(K key, V value) {
+        int bucketIndex = getIndex(key);
+        if (table[bucketIndex] == null) {
+            table[bucketIndex] = new Node<>(key, value, null);
+            size++;
+        } else {
+            Node<K, V> currentNode = table[bucketIndex];
+            while (currentNode.next != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode.next = new Node<>(key, value, null);
+            size++;
```

Comment:
```suggestion
            Node<K, V> currentNode = table[bucketIndex];
            while (currentNode!= null) {
                if (currentNode.key == key
                    || currentNode.key != null && currentNode.key.equals(key)) {
                    currentNode.value = value;
                    return;
                }
                if (currentNode.next == null) {
                    currentNode.next = new Node<>(key, value, null);
                    size++;
                   return;
               }
               currentNode = currentNode.next;
           }
```
sometimes break; operator can confuse a little bit, you can simplify it like this